modules/core/src/main/java/org/apache/synapse/mediators/elementary/Target.java

```diff
@@ -458,6 +463,35 @@ public void insertJson(MessageContext synCtx, Object sourceJsonElement, SynapseL
         }
     }
 
+    /**
+     * This method will remove all the matching elements of the given jsonPath from the JSON payload in the msg
+     * context.
+     *
+     * @param synCtx   message context.
+     * @param jsonPath JSON-path expression to select the removing element.
+     */
+    public void removeJson(MessageContext synCtx, SynapsePath jsonPath)
+            throws IOException, PathNotFoundException {
+        SynapseJsonPath synapseJsonPath = (SynapseJsonPath) jsonPath;
+        Axis2MessageContext axis2smc = (Axis2MessageContext) synCtx;
+        org.apache.axis2.context.MessageContext axis2MessageCtx = axis2smc.getAxis2MessageContext();
+        // handle complete removal of the message body.
+        String jsonPathString = synapseJsonPath.toString();
+        // removing "json-eval(" and extract only the expression
+        jsonPathString = jsonPathString.substring(10, jsonPathString.length() - 1);
+        if (jsonPathString.equals("$") || jsonPathString.equals("$.")) {
```

Comment:
Shall we the isWholeBody function here? https://github.com/wso2/wso2-synapse/blob/master/modules/core/src/main/java/org/apache/synapse/util/xpath/SynapseJsonPath.java#L354